substrate/frame/support/procedural/src/pallet/expand/pallet_struct.rs

```diff
@@ -83,9 +83,7 @@ pub fn expand_pallet_struct(def: &mut Def) -> proc_macro2::TokenStream {
 			impl<#type_impl_gen> #pallet_ident<#type_use_gen> #config_where_clause {
 				#[doc(hidden)]
 				pub fn error_metadata() -> Option<#frame_support::__private::metadata_ir::PalletErrorMetadataIR> {
-					Some(#frame_support::__private::metadata_ir::PalletErrorMetadataIR {
-						ty: #frame_support::__private::scale_info::meta_type::<#error_ident<#type_use_gen>>()
-					})
+					Some(<#error_ident<#type_use_gen>>::error_metadata())
 				}
 			}
```

Comment:
note that you could move this whole impl block inside `expand/error.rs` same as call